Allow backing up of starred (and other) gists

fbartels opened the pull request:
Hi Adam,

just recently found your script and also wanted to backup gists that I have starred. Luckily your script was easy to adapt to this.

Regards Felix

aprescott commented:
@fbartels I'm not ignoring you, I promise, I just haven't managed to find time to get to looking at this properly. I'll try to get to it soon!

fbartels commented:
Hi Adam,

no worries. The change it actually a lot smaller than it looks in the github diff view and I've been running it for a while now.

aprescott commented:
@fbartels I think I'm against dumping all the starred repos in the same directory as a user's own gists. I'm also not sure this is something that should be default behaviour.

What do you think about an `--include-starred` option that places starred gists in `starred/`?

fbartels commented:
Hi,

I understand your objections regarding dumping all gist repos into a single dir. The benefit of the current approach would be, that you can easily add the gist urls of other users to download them as well.

Adding ```--include-starred``` would also be a nice option, I also toyed with the idea to make this configurable with "git config".

aprescott commented:
What if, instead, the URL was configurable through an environment variable? It could default to the existing `https://api.github.com/gists`, but the documentation could specify that you can call it with, say,

```shell
GIST_URL=https://api.github.com/gists/starred gist-backup
```

or any other URL (such as those of a particular user). This would of course rest on the assumption that the GitHub API interaction is identical for all gist-y endpoints.

fbartels commented:
Putting the gist url into an environment variable sounds like a great idea. I just updated my pull request according to your suggestions and also changed its title.

CMCDragonkai commented:
Any more progress on this?

fbartels commented:
You could use my repo (https://github.com/fbartels/gist-backup) the changes for this are complete and fully functional.

CMCDragonkai commented:
Is there no URL that has your own gists and your starred gists? It would save having to switch between the 2 URLs.

fbartels commented:
No, sadly there is no possibility for this in the Gists API. I have created a little wrapper script that calls gist-backup for the gists, and afterwards calls it again with the GIST_URL env.

aprescott commented on gist-backup:
```diff
@@ -97,10 +107,10 @@ retries=0
 while [ $retries -lt $MAX_RETRIES ]
 do
-  echo "Requesting Gist page: $page"
+  echo "Requesting Gist page: $page" from $url
```
There's a minor issue here with the `"`s around the full content, but I can fix that up on `master` after merging.

aprescott commented:
Sorry for the slow progress on this. I like this `GIST_URL` solution a lot! Thanks so much for spending time putting this together, @fbartels.
